src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -337,13 +337,21 @@ protected Parameter[] GetParametersForResourceEntry(ResourceCollection resourceC
 
         protected virtual void WriteResourceCollectionEntry(ResourceCollection resourceCollection, MethodSignature signature)
         {
-            _writer.Append($"return new {resourceCollection.Type.Name}({ArmClientReference}, Id, ");
-            foreach (var parameter in resourceCollection.ExtraConstructorParameters)
+            if (resourceCollection.ExtraConstructorParameters.Count() > 0)
             {
-                _writer.Append($"{parameter.Name}, ");
+                _writer.Append($"return new {resourceCollection.Type.Name}({ArmClientReference}, Id, ");
+                foreach (var parameter in resourceCollection.ExtraConstructorParameters)
+                {
+                    _writer.Append($"{parameter.Name}, ");
+                }
+                _writer.RemoveTrailingComma();
+                _writer.Line($");");
```

Comment:
Shall we add a comment of "TODO" here saying that in the future we will also consider to do caching for the collection with parameters as well? Or we will not do it?

Reply:
done